nncf/quantization/algorithms/bias_correction/algorithm.py

```diff
@@ -468,77 +485,114 @@ def get_statistic_points(self, model: TModel) -> StatisticPointsContainer:
-                    num_samples=self.subset_size, inplace=self.inplace_statistics
-                )
-                statistic_container.add_statistic_point(
-                    StatisticPoint(
-                        target_point=statistic_point, tensor_collector=stat_collector, algorithm=BiasCorrection
-                    )
-                )
+            # We assume that input node has only one output port
+            output_port_id = 0
+            for next_layer in nncf_graph.get_next_nodes(input_node):
+                self._collected_stat_inputs_map[next_layer.node_name] = (input_node.node_name, output_port_id)
+            statistic_point = self._backend_entity.target_point(
+                TargetType.POST_LAYER_OPERATION, input_node.node_name, port_id=output_port_id
+            )
+            stat_collector = self._backend_entity.raw_statistic_collector(
+                num_samples=self.subset_size, inplace=self.inplace_statistics
+            )
+            statistic_container.add_statistic_point(
+                StatisticPoint(target_point=statistic_point, tensor_collector=stat_collector, algorithm=BiasCorrection)
+            )
 
         return statistic_container
 
-    def _get_biased_after_input_nodes(self, nncf_graph: NNCFGraph, model_inputs: List[NNCFNode]) -> Dict[str, str]:
+    def _get_activation_node_and_port(self, node: NNCFNode, nncf_graph: NNCFGraph) -> Tuple[NNCFNode, int]:
+        """
+        This method returns the activation layer and corresponding port id for the node.
+
+        :param node: NNCFGraph node for which the activation is sought.
+        :param nncf_graph: NNCFGraph instance with the node.
+        :return: Tuple with the activation node and port id.
         """
-        This method finds and returns the first nodes with the bias in the model that follows after the input nodes.
+        activation_port = self._backend_entity.get_activation_port_id(node, nncf_graph)
+        activation_node = nncf_graph.get_input_edges(node)[activation_port].from_node
+        port_id = nncf_graph.get_edge(activation_node, node).output_port_id
+        return activation_node, port_id
+
+    def _get_biased_after_nodes(self, nncf_graph: NNCFGraph, nodes: List[NNCFNode], model: TModel) -> List[NNCFNode]:
+        """
+        This method finds and returns nodes with the bias in the model that follows after the input nodes.
 
         :param nncf_graph: NNCFGraph instance.
-        :param model_inputs: List of the model inputs as NNCFNodes.
-        :return: A dictionary with the names of the nodes with bias as keys and their input node names as values.
+        :param nodes: List of the model inputs as NNCFNodes.
+        :param model: TModel instance.
+        :return: List of the nodes with bias.
         """
 
-        def traverse_to_biased(node, output):
-            if node in output:
-                return True, output
+        def traverse_to_biased(node, condition_container):
```

Comment:
It looks like you several times implemented traversal function. Could you think about reusable implementation as separate module to agregate traversal algorithms, for example https://networkx.org/documentation/stable/reference/algorithms/traversal.html?

Reply:
Yes, you are right. It looks like you use depth first searching.  My comment relates to that you several times implement the same recursion in BC algorithm. This comment is not stopper for merging but I suggest to refactoring your algorithm and create separate function for depth first searching.